frame/support/src/inherent.rs

```diff
@@ -20,8 +20,15 @@ pub use crate::sp_std::vec::Vec;
 #[doc(hidden)]
 pub use crate::sp_runtime::traits::{Block as BlockT, Extrinsic};
 #[doc(hidden)]
-pub use sp_inherents::{InherentData, ProvideInherent, CheckInherentsResult, IsFatalError};
+pub use sp_inherents::{
+	InherentData, ProvideInherent, CheckInherentsResult, IsFatalError, InherentIdentifier,
+	MakeFatalError,
+};
 
+/// The identifier for runtime inherent errors.
+///
+/// The associated error is a `String`.
+pub const RUNTIME_INHERENT_IDENTIFIER: InherentIdentifier = *b"RuntimeI";
```

Comment:
I would like to see this check in `execute_block`.

Reply:
done, I now check_inherent only check inherents, the position of inherent is checked in execute_block.

But the check is not done in apply_extrinsics. Should we also do the check for block producer ?

Reply:
No. If someone tries to trick himself by modifying the node, let them do :P This is no problem.